"internal error: environment misses: peerOrResponder" - with latest Nim-1.4-RC

```text
[amd] 25 Wed Oct 07 18:29:27 |/mnt/sda3/storage/CODE/status/nimbus|
stefan$ PATH="/src/77_DLD/CODE/00_github/Nim/bin:$PATH" make USE_SYSTEM_NIM=1
Building: build/test_tools_build
[using system Nim: /src/77_DLD/CODE/00_github/Nim/bin/nim]
/mnt/sda3/storage/CODE/status/nimbus/vendor/nim-eth/eth/p2p/rlpx.nim(817, 23) Error: internal error: environment misses: peerOrResponder
make: *** [Makefile:79: test_tools_build] Error 1
```

https://github.com/status-im/nim-eth/blob/de2d43a7e7afb7b094ca251bdbbd58fbf47df031/eth/p2p/rlpx.nim#L817

https://github.com/nim-lang/nightlies/releases/tag/2020-10-07-version-1-4-3b901d1e361f49d48fb64d115e42c04a4a37100c (just a "devel" snapshot)

--- a/eth/p2p/p2p_protocol_dsl.nim
+++ b/eth/p2p/p2p_protocol_dsl.nim
@@ -247,6 +247,13 @@ proc outputParamType*(msg: Message): NimNode =
   if outputParam != nil:
     return outputParam[1]
 
+proc refreshParam(n: NimNode): NimNode =
+  result = copyNimTree(n)
+  if n.kind == nnkIdentDefs:
+    for i in 0..<n.len-2:
+      if n[i].kind == nnkSym: 
+        result[i] = genSym(symKind(n[i]), $n[i])
+
 iterator typedInputParams(procDef: NimNode, skip = 0): (NimNode, NimNode) =
   for paramName, paramType in typedParams(procDef, skip):
     if not isOutputParamName(paramName):
@@ -256,7 +263,7 @@ proc copyInputParams(params: NimNode): NimNode =
   result = newTree(params.kind)
   for param in params:
     if not isOutputParam(param):
-      result.add param
+      result.add param.refreshParam
 
 proc chooseFieldType(n: NimNode): NimNode =
   ## Examines the parameter types used in the message signature
@@ -790,6 +797,7 @@ proc createHandshakeTemplate*(msg: Message,
   forwardCall[1] = peerVar
   forwardCall.del(forwardCall.len - 1)
 
+  let peerVar = genSym(nskLet ,"peer")
   handshakeExchanger.setBody quote do:
     let `peerVar` = `peerValue`
     let sendingFuture = `forwardCall`

